File: Python/septimo_ejercicio.py
import random
from utils import es_numero_entero
from rich.console import Console
console = Console()
def calcular_frecuencia(letras=[]):
    frecuencia=[0]*26
    for letra in letras:
        frecuencia[ord(letra) - ord('a')] +=1
    console.print("[bold blue]La frecuencia de cada caracter: [bold blue]")
    for i in range(len(frecuencia)):
        console.print(f"[bold red]{chr(ord('a') + i)+": "+str(frecuencia[i])}[bold red]")

def cant_letras():
    letras = []
    cantidad = es_numero_entero(console.input("[italic magenta]Ingrese la cantidad de letras \n[italic magenta]"))
    for i in range(cantidad):
        letra = chr(ord('a') + int(random.random() * (ord('z') - ord('a') + 1)))
        letras.append(letra)

    console.print("[italic green]El vector es: [italic green]")
    for letra in letras:
        print(letra)
    console.print("[italic yellow]\nFrecuencia de letras:  [italic yellow]")
    calcular_frecuencia(letras)

if __name__ == "__main__":
    cant_letras()

File: Java/septimoEjercicio.java
import java.util.Scanner;

public class septimoEjercicio {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Ingrese la cantidad de letras ");
        int cantidad = esNumero(scanner);
        char[] letras = new char[cantidad];
        for (int i =0; i<cantidad;i++){
            letras[i]= (char) ('a'+ Math.random()*('z' - 'a'+1));
        }
        System.out.println("El vector es: ");
        for (char letra: letras){
            System.out.println(letra + " ");
        }
        System.out.println(" ");
        calcularFrecuencia(letras);
    }
    public static int esNumero(Scanner scanner){
        while (true){
            try {
                return Integer.parseInt(scanner.next());
            }catch (NumberFormatException e){
                System.out.println("Error: Debe ingresar un número.");
            }
        }
    }
    public static void calcularFrecuencia(char[] letras){
        int[] frecuencia = new int[26];
        for (char letra : letras){
            frecuencia[letra - 'a']++;
        }
        System.out.println("La frecuencia dec cada caracter");
        for (int i=0; i<frecuencia.length;i++){
            System.out.println((char) ('a'+i) + ": "+frecuencia[i]);
        }
    }
}
